model/task/task.go

```diff
@@ -1745,6 +1740,108 @@ func (t *Task) Archive() error {
 	return nil
 }
 
+func ArchiveMany(tasks []Task) error {
+	if len(tasks) == 0 {
+		return nil
+	}
+	// add execution tasks of display tasks passed in, if they are not there
+	execTaskMap := map[string]*Task{}
```

Comment:
since you only use this to check if the exec task exists, map[string]bool would work